tests/Feature/TagControllerTest.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace Tests\Feature\Api;
+
+use App\Models\Tag;
+use App\Models\User;
+use Database\Factories\TagFactory;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Spatie\Permission\Models\Role;
+use Tests\TestCase;
+
+
+class TagControllerTest extends TestCase
+{
+    use RefreshDatabase;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $user = User::factory()->create();
+        $adminRole = Role::where('name', 'admin')->first();
+
+        if (!$adminRole) {
+            // If the "admin" role does not exist, create it
+            $adminRole = Role::create(['name' => 'admin']);
+        }
+
+        $user->assignRole($adminRole);
+        $this->actingAs($user, 'api');
+    }
+
+    public function testIndexReturnsTags()
+    {
+        // Create some tags
+        TagFactory::new()->count(3)->create();
+
+        $response = $this->json('GET', '/api/v1/tags');
+
+        $response->assertStatus(200);
+
+        $response->assertJsonStructure([
+            'data' => [
+                '*' => [
+                    'id',
+                    'tag_name',
+                ],
+            ],
+        ]);
+    }
+
+    public function testIndexReturns404WhenNoTagsExist()
+    {
+        $response = $this->json('GET', '/api/v1/tags');
+
+        $response->assertStatus(404);
+
+        $response->assertJson(['message' => 'No hi ha tags a la base de dades']);
+    }
```

Comment:
This is related with the comment in `TagController.php` in L18
```suggestion
        $response->assertJson(['message' => __('Not Found')]);
```